helpers/cpu.rb

```diff
@@ -27,7 +27,16 @@
 def cpus
   def create_cpu(cpu_hash)
     OpenStruct.new(cpu_hash).tap do |o|
-      o.model = cpu_hash['model'] || cpu_hash['version'] || 'No model found'
+      o.model = cpu_hash['model'] || cpu_hash['version'] || cpu_hash['product'] || 'No model found'
+
+      # Determine total number of cores
+      ## For metal systems: Uses the "enabledcores" value from lscpu
+      ## For VMs: Each CPU is 1 core (and no enabledcores setting exists)
+      unless o['configuration'].nil?
+        o.cores = find_hashes_with_key_value(o['configuration']['setting'], 'id', 'enabledcores')[0]['value'].to_i
```

Comment:
How confident are you that if `o['configuration']` is present, its value will be in the correct format?  Because if it isn't, the `find_hashes_with_key_value(...)[0]['value']`  could result in a render error.

Something like the following would be "safer":

```
o.cores = 1
unless o['configuration'].nil?
  hashes = find_hashes_with_key_value(o['configuration']['setting'], 'id', 'enabledcores')
  unless hashes.empty?
    o.cores = [0]['value']
  end
end
```

Its "safer" in that its less likely to result in a render error.  But perhaps if the format of `o['configuration']` has changed, we'd want to the tool to shout about it loudly?

Reply:
I appreciate the worry regarding the configuration parameter. I'm confident that it should be consistent and agree that, should it change, that errors should be thrown! Because of this, I'll skip this change.